sherpa/astro/data.py

```diff
@@ -1523,23 +1523,43 @@ def _get_rate(self):
     def _set_rate(self, val):
         self._rate = bool_cast(val)
         for id in self.background_ids:
-            # TODO: shouldn't this store bool_cast(val) instead?
             self.get_background(id).rate = val
 
     rate = property(_get_rate, _set_rate,
-                    doc='Quantity of y-axis: counts or counts/sec')
+                    doc="""Is the Y axis displayed as a rate when plotting data?
+
+When True the y axis is normalised by the exposure time to display
+a rate.""")
 
     def _get_plot_fac(self):
         return self._plot_fac
 
     def _set_plot_fac(self, val):
-        self._plot_fac = int(val)
+        # I'd prefer to check whether val is an integer, but there
+        # may be users who have set the value to 2.0 and it doesn't
+        # seem worth breaking that code.
+        #
+        try:
+            okay = val == int(val)
+        except (ValueError, TypeError):
+            # For when int() can't convert val, which can raise
+            # different errors.
+            okay = False
+
+        if not okay:
+            raise DataErr("bad", "plot_fac setting", val)
+
+        val = int(val)
+        self._plot_fac = val
         for id in self.background_ids:
             self.get_background(id).plot_fac = val
 
     plot_fac = property(_get_plot_fac, _set_plot_fac,
-                        doc='Number of times to multiply the y-axis ' +
-                        'quantity by x-axis bin size')
+                        doc="""How the X axis is used to create the Y axis when plotting data.
+
+The Y axis values are multiplied by X^plot_fac. The default
+value of 0 means the X axis is not used in plots. The value
```

Comment:
That means the plots don't have an x-axis?

Reply:
I was trying to come up with something meaningful to say here, but failed. The problem is that I don't really want to say "the y axis is multiplied by x axis^plot_fac" (even though I do actually say this) as it is logically wrong - the y axis is the y axis. What I'm really trying to say is that the "base units" are multiplied by x^plot_fac values to create the y axis, but there's no definition of "base units" (it depends on the rate setting, for one thing, and the plot type for another). So options include:

- dropping this text, just leaving the first line
- trying to come up with some very basic text that doesn't really say anything (and if so, is it worth keeping)
- coming up with a better way to describe this

Any ideas?

Reply:
How about:
```
'''Scale y values when plotting data

y-axis values can be multiplied by x^plot_fac
where x is the corresponding x-value for each datapoint. 
The default is plot_fac=0, so no scaling is applied. This is useful to show e.g. counts/kev on the y-axis of the plot.
'''
```
(Does my example make any sense?)

Reply:
I've approved it already. I don't think my wording is that much better than what's in the PR now, but feel free to use either or a combination of both.